django_statsd/middleware.py

```diff
@@ -7,34 +8,38 @@
 from django_statsd.clients import statsd
 
 
-class GraphiteMiddleware(object):
+try:
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:  # pragma: no cover
+    MiddlewareMixin = object
+
+
+class GraphiteMiddleware(MiddlewareMixin):
 
     def process_response(self, request, response):
         statsd.incr('response.%s' % response.status_code)
-        if hasattr(request, 'user') and request.user.is_authenticated():
+        if hasattr(request, 'user') and request.user and request.user.is_authenticated():
```

Comment:
Do we really have cases where request.user can be `None` ?

Reply:
I think I cherry-pick't that one in. Figured it wouldn't harm checking if the user is actually something before calling an attribute.
The alternative would be that it throws an ugly 'NoneType doesn't have attribute is_authenticated' as we would assume that `request.user` can't be None.